File: Plugins/Interfaces.vb

''' <summary>
''' Interface host.
''' </summary>
''' <remarks></remarks>
Public Interface Host

    ''' <summary>
    ''' 
    ''' </summary>
    ''' <param name="strFeedback"></param>
    ''' <remarks></remarks>
    Sub ShowFeedback(ByVal strFeedback As String)

End Interface

''' <summary>
''' Interface plugin.
''' </summary>
''' <remarks></remarks>
Public Interface Plugin

    ''' <summary>
    ''' 
    ''' </summary>
    ''' <param name="Host"></param>
    ''' <remarks></remarks>
    Sub Initialize(ByVal Host As Host)

    ''' <summary>
    ''' 
    ''' </summary>
    ''' <value></value>
    ''' <returns></returns>
    ''' <remarks></remarks>
    ReadOnly Property Name() As String

    ''' <summary>
    ''' 
    ''' </summary>
    ''' <param name="Args">Arguments that will be passed to the plugin.</param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Function Main(ByVal Args As Array) As Integer

End Interface

File: Plugins/Interfaces.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using System.Diagnostics;
using System.Data;
using System.Xml.Linq;
using Microsoft.VisualBasic;
using System.Collections;
using System.Linq;
// End of VB project level imports



/// <summary>
/// Interface host.
/// </summary>
/// <remarks></remarks>
namespace Versidyne.Plugins
{
	public interface Host
	{
		
		/// <summary>
		///
		/// </summary>
		/// <param name="strFeedback"></param>
		/// <remarks></remarks>
		void ShowFeedback(string strFeedback);
		
	}
	
	/// <summary>
	/// Interface plugin.
	/// </summary>
	/// <remarks></remarks>
	public interface Plugin
	{
		
		/// <summary>
		///
		/// </summary>
		/// <param name="Host"></param>
		/// <remarks></remarks>
		void Initialize(Host Host);
		
		/// <summary>
		///
		/// </summary>
		/// <value></value>
		/// <returns></returns>
		/// <remarks></remarks>
		string Title {get;}
		
		/// <summary>
		///
		/// </summary>
		/// <param name="Args">Arguments that will be passed to the plugin.</param>
		/// <returns></returns>
		/// <remarks></remarks>
		int Main(Array Args);
		
	}
}
